src/openms/source/METADATA/ID/IdentificationData.cpp

```diff
@@ -190,42 +212,59 @@ namespace OpenMS
     return result.first;
   }
 
-
-  IdentificationData::DataQueryRef
-  IdentificationData::registerDataQuery(const DataQuery& query)
+  IdentificationData::ObservationRef
+  IdentificationData::registerObservation(const Observation& obs)
   {
-    // reference to spectrum or feature is required:
-    if (query.data_id.empty())
+    if (!no_checks_)
     {
-      String msg = "missing identifier in data query";
-      throw Exception::IllegalArgument(__FILE__, __LINE__,
-                                       OPENMS_PRETTY_FUNCTION, msg);
+      // reference to spectrum or feature is required:
+      if (obs.data_id.empty())
+      {
+        String msg = "missing identifier in observation";
+        throw Exception::IllegalArgument(__FILE__, __LINE__,
+                                         OPENMS_PRETTY_FUNCTION, msg);
+      }
+      // ref. to input file must be valid:
+      if (!isValidReference_(obs.input_file, input_files_))
+      {
+        String msg = "invalid reference to an input file - register that first";
+        throw Exception::IllegalArgument(__FILE__, __LINE__,
+                                         OPENMS_PRETTY_FUNCTION, msg);
+      }
     }
-    // ref. to input file may be missing, but must otherwise be valid:
-    if (query.input_file_opt && !isValidReference_(*query.input_file_opt,
-                                                   input_files_))
+
+    // can't use "insertIntoMultiIndex_" because Observation doesn't have the
+    // "steps_and_scores" member (from ScoredProcessingResult)
+    auto result = observations_.insert(obs);
+    if (!result.second) // existing element - merge in new information
     {
-      String msg = "invalid reference to an input file - register that first";
-      throw Exception::IllegalArgument(__FILE__, __LINE__,
-                                       OPENMS_PRETTY_FUNCTION, msg);
+      observations_.modify(result.first, [&obs](Observation& existing)
+                           {
+                             existing.merge(obs);
+                           });
     }
-    DataQueryRef ref = data_queries_.insert(query).first;
-    data_query_lookup_.insert(ref);
-    return ref;
+
+    observation_lookup_.insert(uintptr_t(&(*result.first)));
+
+    // @TODO: add processing step (currently not supported by Observation)
```

Comment:
what could be an application of having a processing step at observation?

Reply:
Not sure, which is why observations currently don't have processing steps. ;-)
I've added a question mark to the comment.

Reply:
How big would be the implications/changes needed if we would need to add this later?

Reply:
Actually, RT transformation is a processing step that gets applied to observations, because that's where the RT information is stored. Currently that is not tracked explicitly, because the observations are considered "meta data", rather than "actual data". It could definitely be changed if we wanted to, but would make each observation "bigger".